src/enigma2/data/ChannelGroup.cpp

```diff
@@ -8,6 +8,39 @@ using namespace enigma2;
 using namespace enigma2::data;
 using namespace enigma2::utilities;
 
+bool ChannelGroup::Like(const ChannelGroup &right) const
+{
+  bool isLike = true;
+  isLike = isLike && (m_serviceReference == right.m_serviceReference);
+  isLike = isLike && (m_groupName == right.m_groupName);
+
+  return isLike;
+}
+
+bool ChannelGroup::operator==(const ChannelGroup &right) const
+{
+  bool isEqual = true;
+  isEqual = isEqual && (m_serviceReference == right.m_serviceReference);
+  isEqual = isEqual && (m_groupName == right.m_groupName);
+  isEqual = isEqual && (m_radio == right.m_radio);
+  isEqual = isEqual && (m_lastScannedGroup == right.m_lastScannedGroup);
+
+  for (int i = 0; i < m_channelList.size(); i++)
```

Comment:
using a range-based loop here would be more "sexy", espacially as ugly `at(i)` calls are not needed, then.

Reply:
But then how would I pull the channels from two different vectors? If I had one vector that would make sense. Or I'm missing something?

Reply:
Ah, I see. Overlooked m_chanellist vs. right.m_channellist. nvm.